src/lib.rs

```diff
@@ -239,3 +242,59 @@ pub trait ReadFrom {
     where
         Self: Sized;
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_test::{assert_de_tokens, assert_de_tokens_error, Token};
+
+    #[test]
+    fn network_can_deserialize() {
+        assert_de_tokens(
+            &Network::MainNet,
+            &[Token::UnitVariant {
+                name: "Network",
+                variant: "mainnet",
+            }],
+        );
+        assert_de_tokens(
+            &Network::TestNet,
+            &[Token::UnitVariant {
+                name: "Network",
+                variant: "testnet",
+            }],
+        );
+        assert_de_tokens_error::<Network>(
+            &[Token::UnitVariant {
+                name: "Network",
+                variant: "other",
+            }],
+            "unknown variant `other`, expected `mainnet` or `testnet`",
+        );
+    }
+
+    #[test]
+    fn keytype_can_deserialize() {
```

Comment:
The clippy errors have been fixed, now revealing that this test fails.

Reply:
How is this test run? `cargo test` doesn’t fail at all for me

Reply:
Nvm, I see it’s running features-specific tests with a flag